server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -118,17 +118,24 @@ public RemoteStoreRefreshListener(
         resetBackOffDelayIterator();
         this.checkpointPublisher = checkpointPublisher;
         this.statsListener = new UploadListener() {
+            private long uploadStartTime = 0;
+
             @Override
             public void beforeUpload(String file) {
                 // Start tracking the upload bytes started
                 segmentTracker.addUploadBytesStarted(segmentTracker.getLatestLocalFileNameLengthMap().get(file));
+                uploadStartTime = System.currentTimeMillis();
             }
 
             @Override
             public void onSuccess(String file) {
                 // Track upload success
                 segmentTracker.addUploadBytesSucceeded(segmentTracker.getLatestLocalFileNameLengthMap().get(file));
                 segmentTracker.addToLatestUploadedFiles(file);
+                long fileUploadTime = System.currentTimeMillis() - uploadStartTime;
+                // Round off upload time to 1 millisecond
+                fileUploadTime = fileUploadTime > 0 ? fileUploadTime : 1;
```

Comment:
This code is not thread safe.

Reply:
As discussed, creating the `UploadListener` object per file upload to ensure that the operations are thread-safe